transmogrifier/sources/xml/marc.py

```diff
@@ -32,109 +33,41 @@
 class Marc(XMLTransformer):
     """Marc transformer."""
 
-    def get_optional_fields(self, xml: Tag) -> dict | None:
+    def get_optional_fields(self, source_record: Tag) -> dict | None:
         """
         Retrieve optional TIMDEX fields from a MARC XML record.
 
         Overrides metaclass get_optional_fields() method.
 
         Args:
-            xml: A BeautifulSoup Tag representing a single MARC XML record.
+            source_record: A BeautifulSoup Tag representing a single MARC XML record.
         """
         fields: dict = {}
 
-        source_record_id = Marc.get_source_record_id(xml)
-
-        fixed_length_data = xml.find("controlfield", tag="008", string=True)
-        if fixed_length_data is None:
-            message = f"Record ID {source_record_id} is missing MARC 008 field"
-            logger.error(message)
-            return None
+        source_record_id = Marc.get_source_record_id(source_record)
 
-        leader = xml.find("leader", string=True)
-        if leader is None:
-            message = f"Record ID {source_record_id} is missing MARC leader"
-            logger.error(message)
-            return None
+        leader_field = Marc._get_leader_field(source_record)
+        control_field_general_info = Marc._get_control_field_general_info(source_record)
```

Comment:
I see your thinking a bit in having these temporarily here for use in field that don't have their own methods yet... but what about just calling these new and/or renamed methods from those places directly?

e.g. I see `leader_field` used, but could call directly:
```python
# content_type
if content_type := Marc.json_crosswalk_code_to_name(
    self._get_leader_field(source_record)[6:7],
    marc_content_type_crosswalk,
    source_record_id,
    "Leader/06",
):
    fields["content_type"] = [content_type]
```

What I think this might help guide is how often these are used.  If often, it might be worth caching the results on those sub-methods.  If not often, it's probably okay to invoke them directly 1-2 times.  I'm a big fan of caching -- where you call the method, and it reuses data if safe to cache and already called -- but there is a bit of overhead there.

Either way, might avoid needing these temporary values at all, and would give some insight into usage during future parts of this refactor.   Just a thought.

Reply:
Updated `Marc` to call the private methods as needed but as discussed, caching is not needed at this time!